### PR TITLE
fix(sdk): podSpecPatch bug. Fixes 6512

### PR DESCRIPTION
**Pull request by NikeNano**
**Description of your changes:**

This PR fixes: https://github.com/kubeflow/pipelines/issues/6512

The bug is found here: 

https://github.com/kubeflow/pipelines/blob/19b3de4dd049052de74868ad0dfc931a4c700265/sdk/python/kfp/compiler/_op_to_template.py#L321

Will spin up a GKE cluster tonight to validate using GPU:s

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by chensun** [comment]
@NikeNano, thanks for the fix. Can you please add an item under: https://github.com/kubeflow/pipelines/blob/master/sdk/RELEASE.md#bug-fixes-and-other-changes

**Comment by NikeNano** [comment]
> @NikeNano, thanks for the fix. Can you please add an item under: https://github.com/kubeflow/pipelines/blob/master/sdk/RELEASE.md#bug-fixes-and-other-changes

Updated.

**Comment by NikeNano** [comment]
I could be that we get the timeout out due to the lack of GPU resources but not sure. I will test to remove the test and see if it works.

File: samples/test/config.yaml [hunk 1]
```diff
@@ -42,6 +42,8 @@
   path: samples.core.resource_spec.resource_spec_test
 - name: runtime_resource_spec
   path: samples.core.resource_spec.runtime_resource_request_test
+- name: runtime_resource_spec_gpu
+  path: samples.core.resource_spec.runtime_resource_request_gpu_test
```
Comment (NikeNano) [inline samples/test/config.yaml:46]:
```suggestion
```

**Comment by NikeNano** [comment]
So it seems like we have to either remove the test @chensun or increase the resource quota. I leave this for you do decide upon. Let me know and I will update the PR accordingly.

**Comment by chensun** [comment]
> So it seems like we have to either remove the test @chensun or increase the resource quota. I leave this for you do decide upon. Let me know and I will update the PR accordingly.

It's fine to remove the test. 

/lgtm

**Pull request merged**